Transformer (Encoder)/train_model.py

```diff
@@ -0,0 +1,414 @@
+"""This file implements the pipeline for training the model.
+
+This file defines a complete pipline to train to train and test
+a transformer-based model to predict intents and slots on the ATIS
+ans Snips datasets. Some of the code was in this file was taken
+from https://www.tensorflow.org/tutorials/text/transformer.
+
+  Typical usage example:
+
+    python train_model.py --dataset='atis'
+"""
+
+import argparse
```

Comment:
I used to use click module https://click.palletsprojects.com/en/7.x/ for this, let me know what do you think.